WooCommerce/Classes/ViewRelated/Dashboard/ProductStock/ProductStockDashboardCardViewModel.swift

```diff
@@ -154,7 +155,14 @@ private extension ProductStockDashboardCardViewModel {
                         productIDs: Array(Set(variationsToFetchReports.map { $0.parentID })),
                         variationIDs: variationsToFetchReports.map { $0.productID }
                     )
-                    return self.updatedVariationReports(from: reports, using: variationsToFetchReports)
+                    return reports.map { report in
+                        guard let variationID = report.variationID,
+                              report.productID == 0,
+                              let parentID = stock.first(where: { $0.productID == variationID })?.productID else {
+                            return report
+                        }
+                        return report.copy(productID: parentID)
+                    }
```

Comment:
Referencing the main actor-isolated `self` in an async context is tricky, so I moved the logic back to the async group instead of using the helper method.